main.go

```diff
@@ -186,3 +192,13 @@ func usage() {
 	os.Stdout.WriteString("<providerName>: optional, DNSCrypt provider name\n")
 	os.Stdout.WriteString("<serverPk>: optional, DNSCrypt server public key\n")
 }
+
+// newEDNS0Padding constructs a new OPT RR EDNS0 Padding for the extra section.
+func newEDNS0Padding() (extra dns.RR) {
+	return &dns.OPT{
+		Hdr: dns.RR_Header{Name: ".", Rrtype: dns.TypeOPT, Class: 2000},
+		Option: []dns.EDNS0{
+			&dns.EDNS0_PADDING{Padding: []byte{0, 0, 0}},
```

Comment:
As per [RFC-8467, sec. 4.1.](https://www.rfc-editor.org/rfc/rfc8467.html#section-4.1):

 >  Clients SHOULD pad queries to the closest multiple of 128 octets.

Thus, length of a whole message should be considered in padding's length calculation, and not just use 3 octets.

Reply:
@Mizzick you should take the whole message length and then add padding option so that the whole length was `N * 128`

Reply:
done